#VRML_SIM R2021a utf8
WorldInfo {
  info [
    "Arena for Stage 1 of SLRC2021 University Category"
  ]
  title "SLRC2021 UniS1"
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9885705432126917 -0.08445409766000068 0.12488309125177291 1.5338551964491993
  position 0.14842346656094108 3.411542946119576 0.23023358694085075
}
DirectionalLight {
  direction 0 -1 0
}
DirectionalLight {
  direction -1 -1 0
}
DirectionalLight {
  direction 0 -1 -1
}
DirectionalLight {
  direction 1 -1 0
}
DirectionalLight {
  direction 0 -1 1
}
RectangleArena {
  translation -0.02 0 0
  name "Floor_black"
  floorSize 3 4
  floorTileSize 2 2
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
  }
  wallAppearance DEF Wall_white PBRAppearance {
    roughness 1
    metalness 0
    name "Wall_white"
  }
}
DEF Line Solid {
  translation -0.613 -0.00485 -1.5
  scale 0.01 0.01 0.01
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Mesh {
        url [
          "../solidworks/Lines.STL"
        ]
      }
      castShadows FALSE
      isPickable FALSE
    }
  ]
  name "White_lines"
  locked TRUE
  translationStep 0
  rotationStep 0
}
DEF Squares Group {
  children [
    DEF Patch Solid {
      translation -0.075 -0.00484 0.37
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "color_patch"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Green Solid {
      translation 0.01 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "green_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Red Solid {
      translation -0.41 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "red_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
    DEF Blue Solid {
      translation 0.43 -0.00484 -1.65
      scale 0.01 0.01 0.01
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Mesh {
            url [
              "../solidworks/Square.STL"
            ]
          }
          castShadows FALSE
          isPickable FALSE
        }
      ]
      name "blue_square"
      locked TRUE
      translationStep 0
      rotationStep 0
    }
  ]
}
DEF Inner_wall Solid {
  translation 0 0 -0.66
  scale 0.01 0.01 0.01
  children [
    DEF Inner_wall Shape {
      appearance USE Wall_white
      geometry Mesh {
        url [
          "../solidworks/Inner_walls.STL"
        ]
      }
    }
  ]
  name "Inner_walls"
  boundingObject USE Inner_wall
  locked TRUE
  recognitionColors [
    1 1 1
  ]
  translationStep 0
  rotationStep 0
}
DEF Right_pillar Solid {
  translation 0.16 0.02 0.74
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.04 0.12
      }
    }
  ]
  name "R_pillar"
  radarCrossSection 1
}
DEF Left_pillar Solid {
  translation -0.16 0.04 0.74
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.09 0.12
      }
    }
  ]
  name "L_pillar"
  radarCrossSection 1
}
Robot {
  translation 0.610002 0 -0.67
  rotation 0 1 0 3.14159
  children [
    DEF body Transform {
      translation 0 0.02 0
      rotation 0 1 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.815686 1 0.388235
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.12 0.01 0.1
          }
        }
      ]
    }
    DEF right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 1432.3199093411852
        anchor -0.055 0.02 -0.03
      }
      device [
        RotationalMotor {
          name "left_motor"
        }
      ]
      endPoint Solid {
        translation -0.055 0.02 -0.0299996
        rotation 0.26607206190288635 0.2660761325776102 -0.9265037234395168 1.6470578970598813
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0.333333 0
              roughness 0.2
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 1432.3199085819226
        anchor 0.055 0.02 -0.03
      }
      device [
        RotationalMotor {
          name "right_motor"
        }
      ]
      endPoint Solid {
        translation 0.055 0.019997 -0.0299951
        rotation -0.28996047152876553 -0.2899642862418821 0.9120546242715136 4.620466204661719
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 0.333333 0
              roughness 0.2
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics Physics {
        }
      }
    }
    DEF support_ball Solid {
      translation 0 0.0075 0.05
      rotation 0 0 -1 2.3276357876976306e-07
      children [
        DEF ball Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.0075
          }
        }
      ]
      name "solid(2)"
      boundingObject USE ball
      physics Physics {
      }
    }
    DEF left_1 DistanceSensor {
      translation 0.0075 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left_1"
      type "infra-red"
    }
    DEF right_1 DistanceSensor {
      translation -0.0075 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right_1"
      type "infra-red"
    }
    DEF right_2 DistanceSensor {
      translation -0.0175 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right_2"
      type "infra-red"
    }
    DEF right_3 DistanceSensor {
      translation -0.0275 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right_3"
      type "infra-red"
    }
    DEF right_4 DistanceSensor {
      translation -0.0375 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "right_4"
      type "infra-red"
    }
    DEF left_2 DistanceSensor {
      translation 0.0175 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left_2"
      type "infra-red"
    }
    DEF left_3 DistanceSensor {
      translation 0.0275 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left_3"
      type "infra-red"
    }
    DEF left_4 DistanceSensor {
      translation 0.0375 0.02 0.06
      rotation 0 0 1 -1.5707
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0235294 0.0352941 0.85098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.008 0.008 0.008
          }
        }
      ]
      name "left_4"
      type "infra-red"
    }
  ]
  boundingObject USE body
  physics Physics {
    density -1
    mass 1
  }
  controller "my_turn_90"
  controllerArgs [
    ""
  ]
}
